lms/services/http.py

```diff
@@ -1,20 +1,44 @@
-import requests
-from requests import RequestException
+from requests import RequestException, Response, Session
 
 from lms.services.exceptions import ExternalRequestError
 
 
 class HTTPService:
     """Send HTTP requests with `requests` and receive the responses."""
 
-    def __init__(self, _session=None):
-        # A requests session is used so that cookies are persisted across
+    session: Session = None
+    """The underlying requests Session."""
```

Comment:
Is this for auto specing it only? I thinks we should mention in on the doc string. Otherwise it's a bit confusing to have class and instance variables with the same name.

Reply:
Yes it was for that. I'll add a comment